src/dlgTriggerEditor.cpp

```diff
@@ -3888,69 +4513,88 @@ void dlgTriggerEditor::saveScript()
         TScript * pT = mpHost->getScriptUnit()->getScript( triggerID );
         if( pT )
         {
-            old_name = pT->getName();
-            if( ! pT->mModuleMasterFolder )
-            {
-                pItem->setText( 0, name );
-                pT->setName( name );
-                pT->setEventHandlerList( handlerList );
-                pT->setScript( mpSourceEditor->toPlainText() );
-                pT->compile();
-            }
-
             QIcon icon;
-            if( pT->isFolder() )
+            if( pT->state() )
             {
-                if( pT->mModuleMasterFolder )
+                old_name = pT->getName();
+                if( ! pT->mModuleMasterFolder )
                 {
-                    if( pT->isActive() )
-                    {
-                        icon.addPixmap( QPixmap( QStringLiteral( ":/icons/folder-brown.png" ) ), QIcon::Normal, QIcon::Off );
-                    }
-                    else
+                    pItem->setText( 0, name );
+                    pT->setName( name );
+                    pT->setEventHandlerList( handlerList );
+                    pT->setScript( mpSourceEditor->toPlainText() );
```

Comment:
:information_source: This __was__ another instance of the bug where a faulty Lua script could not be replaced - it, too, has been fixed in a later commit.